Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -476,29 +476,43 @@ protected override void ProcessRecord()
     [Cmdlet(VerbsCommunications.Read, "GcsObject")]
     public class ReadGcsObjectCmdlet : GcsObjectCmdlet
     {
+        private class ParameterSetNames
+        {
+            public const string ByName = "ByName";
+            public const string ByObject = "ByObject";
+        }
         /// <summary>
         /// <para type="description">
         /// Name of the bucket containing the object. Will also accept a Bucket object.
         /// </para>
         /// </summary>
-        [Parameter(Position = 0, Mandatory = true)]
-        [PropertyByTypeTransformationAttribute(Property = "Name", TypeToTransform = typeof(Bucket))]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ParameterSetNames.ByName)]
+        [PropertyByTypeTransformation(Property = "Name", TypeToTransform = typeof(Bucket))]
         public string Bucket { get; set; }
 
         /// <summary>
         /// <para type="description">
         /// Name of the object to read.
         /// </para>
         /// </summary>
-        [Parameter(Position = 1, Mandatory = true)]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = ParameterSetNames.ByName)]
         public string ObjectName { get; set; }
 
         /// <summary>
         /// <para type="description">
+        /// The Google Cloud Storage bucket object to read.
```

Comment:
Cloud Storage *object* right? Also, might clarify what types it could be. (String or Object object, right?)

"Cloud Storage object to read. May be a Storage Object instance or the name of a storage object (which requires the Bucket parameter be set as well)."